toolkit/tools/scheduler/schedulerutils/graphbuildstate.go

```diff
@@ -147,7 +149,7 @@ func (g *GraphBuildState) RecordBuildResult(res *BuildResult) {
 
 	if !res.Skipped && !res.UsedCache {
 		for _, file := range res.BuiltFiles {
-			if g.isConflictWithToolchain(file) {
+			if !g.supressConflictingRPMs && g.isConflictWithToolchain(file) {
```

Comment:
I'd move `!g.supressConflictingRPMs` to the `if` statement two lines above - we don't need to get into this loop at all if `!g.supressConflictingRPMs` is true.